include/zephyr/posix/signal.h

```diff
@@ -12,6 +12,57 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_POSIX_SIGNAL
+#define SIGHUP    1  /**< Hangup */
+#define SIGINT    2  /**< Interrupt */
+#define SIGQUIT   3  /**< Quit */
+#define SIGILL    4  /**< Illegal instruction */
+#define SIGTRAP   5  /**< Trace/breakpoint trap */
+#define SIGABRT   6  /**< Aborted */
+#define SIGBUS    7  /**< Bus error */
+#define SIGFPE    8  /**< Arithmetic exception */
+#define SIGKILL   9  /**< Killed */
+#define SIGUSR1   10 /**< User-defined signal 1 */
+#define SIGSEGV   11 /**< Invalid memory reference */
+#define SIGUSR2   12 /**< User-defined signal 2 */
+#define SIGPIPE   13 /**< Broken pipe */
+#define SIGALRM   14 /**< Alarm clock */
+#define SIGTERM   15 /**< Terminated */
+/* 16 not used */
+#define SIGCHLD   17 /**< Child status changed */
+#define SIGCONT   18 /**< Continued */
+#define SIGSTOP   19 /**< Stop executing */
+#define SIGTSTP   20 /**< Stopped */
+#define SIGTTIN   21 /**< Stopped (read) */
+#define SIGTTOU   22 /**< Stopped (write) */
+#define SIGURG    23 /**< Urgent I/O condition */
+#define SIGXCPU   24 /**< CPU time limit exceeded */
+#define SIGXFSZ   25 /**< File size limit exceeded */
+#define SIGVTALRM 26 /**< Virtual timer expired */
+#define SIGPROF   27 /**< Profiling timer expired */
+/* 28 not used */
+#define SIGPOLL   29 /**< Pollable event occurred */
+/* 30 not used */
+#define SIGSYS    31 /**< Bad system call */
+
+#define SIGRTMIN 32
+#define SIGRTMAX (SIGRTMIN + CONFIG_POSIX_RTSIG_MAX)
+#define _NSIG (SIGRTMAX + 1)
+
+BUILD_ASSERT(CONFIG_POSIX_RTSIG_MAX >= 0);
```

Comment:
POSIX says 8; that could get checked here or Kconfig, or just stick a note in Kconfig indicating what will make the system POSIX compliant.

Reply:
I've added `Kconfig.limits` to specify the default limit to be posix compliant, but user can freely changes this Kconfig to be anything >= 0, let me know what you think about this approach

These Kconfigs serve as direct translation when we implement `limits.h` in the future

cc @cfriedt

Reply:
That sounds perfect -- tell people how to make POSIX compliant systems, but also allow them to reduce resources if desired.